### PR TITLE
Update leverage calc

### PR DESCRIPTION
**Pull request by dospore**
### Motivation

Leverage calculation was innacurate when tradePrice !== fairPrice

### Changes

- update the leverage calculation

### CODE REVIEW

File: context/OrderContext.tsx [hunk 1]
```diff
@@ -137,7 +137,7 @@ export type OrderState = {
     wallet: number; // ID of corresponding wallet in use 0 -> web3, 1 -> TCR margin
     // boolean to tell if the amount to buy or amount to pay inputs are locked. eg
     //  by changing the amount to pay field it should update the amount to buy and vice versa.
-    //  The lock helps avoiding infinite loops when setting these values
+    //  The lock helps avoiding infinite loops0.18473143662549937 when setting these values
```
Comment (stevenyyan) [inline context/OrderContext.tsx]: what is this number?

Comment (dospore) [reply context/OrderContext.tsx]:
```suggestion
    //  The lock helps avoiding infinite loops when setting these values
```

Comment (dospore) [reply context/OrderContext.tsx]: accidental copy pasta ty

File: context/OrderContext.tsx [hunk 2]
```diff
@@ -280,22 +277,24 @@ export const OrderStore: React.FC<Children> = ({ children }: Children) => {
...
-                    ) {
-                        targetExposure = notional.div(fairPrice);
-                        difference = targetExposure.abs().plus(base.abs());
+                const targetLeverage = new BigNumber(action.leverage);
+                let addedExposure;
+                if (targetLeverage.eq(0)) {
+                    addedExposure = base.abs();
+                } else {
+                    // for long
+                    const x = fairPrice.div(targetLeverage).minus(fairPrice);
```
Comment (dospore) [inline context/OrderContext.tsx:286]:
```suggestion
// set constant x to simplify equation
                    const x = fairPrice.div(targetLeverage).minus(fairPrice);
```

**Review by stevenyyan — DISMISSED** [review]

**Review by stevenyyan — APPROVED** [review]

**Pull request merged**